### PR TITLE
First attempt at updating to GL43C

### PR DESCRIPTION
**Pull request by invalidCards**
Needs more work; most functionality has been restored and switched to using GL43C project-wide but the chunks in the world are not being shaded yet.

@slaytostay Feel free to continue editing on this PR - you probably know a bit more about all the GPU stuff so I'll leave it to you :smile: (plus the dependency update was accepted anyway so it's in no rush)

### CODE REVIEW

File: src/main/java/com/gpu/RegionGpuPlugin.java [hunk 1]
```diff
@@ -180,22 +146,21 @@ public class RegionGpuPlugin extends Plugin implements DrawCallbacks
 	static final String WINDOWS_VERSION_HEADER = "#version 430\n";
 
 	static final Shader PROGRAM = new Shader()
-			.add(GL4.GL_VERTEX_SHADER, "vert.glsl")
-			.add(GL4.GL_GEOMETRY_SHADER, "geom.glsl")
```
Comment (slaytostay) [inline src/main/java/com/gpu/RegionGpuPlugin.java:184]: The gray region shader still uses the geometry shader, so this should also be ported to GL43C

**Review by slaytostay — CHANGES_REQUESTED** [review]
Looks great! I think only the geom shader is missing

**Comment by invalidCards** [comment]
@slaytostay the geometry shader fails (syntax error) using GL43C - looks like you custom wrote it, so I'll leave the fixing of that to you x)

**Comment by slaytostay** [comment]
The shader is originally from the GPU plugin, which they removed at some point, but heavily modified. I won't have a place to work on it for a bit, but it almost sounds like GL43C doesn't support geometry shaders. Which would be a shame, because that would mean we'd have to go back to the fuzzy gray border gradient between regions :/

**Comment by invalidCards** [comment]
We might have to pull in and use the [geometry shader extension](https://javadoc.lwjgl.org/org/lwjgl/opengl/EXTGeometryShader4.html), as they did for the [anisotropic texture filtering extension](https://github.com/slaytostay/region-locker/pull/20/files#diff-2ad705726d6d072adb841117bd77de6293c332f1459f2cfb0f9332d504c46b37R109) in their update as well.

**Comment by slaytostay** [comment]
Ah great, that might work. Is `org.lwjgl.opengl.EXTGeometryShader4` already importable? I might have some time to look at it tonight depending on what the weather does, but if so it might be pretty simple.

**Comment by invalidCards** [comment]
Yep all of `org.lwjgl.opengl` is importable, including `EXTGeometryShader4` 🙂

**Comment by aHooder** [comment]
Thanks for the effort, but I'm closing this in favour of #31

**Pull request closed without merging**